dbms/src/Storages/getStructureOfRemoteTable.cpp

```diff
@@ -31,9 +32,42 @@ ColumnsDescription getStructureOfRemoteTable(
     const Context & context,
     const ASTPtr & table_func_ptr)
 {
-    /// Send to the first any remote shard.
-    const auto & shard_info = cluster.getAnyShardInfo();
+    const auto & shards_info = cluster.getShardsInfo();
 
+    std::string fail_messages;
+
+    for (auto & shard_info : shards_info)
+    {
+        try {
+            const auto & res = getStructureOfRemoteTableInShard(shard_info, database, table, context, table_func_ptr);
+            return res;
+        }
+        catch (const DB::NetException & e)
+        {
+            /// THIS DOES NOT WORK AND IS ALWAYS FALSE???
+            if (context.getSettingsRef().skip_unavailable_shards)
```

Comment:
This is always 0, what I'm missing?

Reply:
The problem is that this is `context.query_context` which is built before query parsing by `context.makeQueryContext()` call in executeQuery.cpp. Not sure what is the right fix.